### ISSUE TITLE
Is `zls` performing any transformation on `_`?

### ISSUE THREAD

**Issue by xphoniex** [zigtools/zls#1390]
![Screenshot from 2023-08-07 08-54-26](https://github.com/zigtools/zls/assets/14107758/9f53f055-ebf9-40f2-a434-9f2ff69bd550)

As you can see, the `_` is orange in the returned highlight response, I'm not sure I've seen this in other lsps. Is there any formatting/transformation happening on underlines?

**Comment by Techatrix** [comment]
I don't think there is any special handling for underscores. If your editor supports markdown lsp responses (hard to tell from your screenshot) this may be because underscores should be escaped in markdown. If you can provide me with some information on your editor setup/config so that I can reproduce this issue, I may be able to more reliably figure out the underlying issue.

**Comment by xphoniex** [comment]
I don't think this is markdown, as I didn't have to touch markdown code, when changing highlights before. anyway I'm on vim8 with https://github.com/ycm-core/YouCompleteMe/commit/3ededaed2f9923d50bf3860ba8dace0f7d2724cd.
